docker/types/services.py

```diff
@@ -0,0 +1,181 @@
+
+    @property
+    def container_spec(self):
+        return self.get('ContainerSpec')
+
+    @property
+    def resources(self):
+        return self.get('Resources')
+
+    @property
+    def restart_policy(self):
+        return self.get('RestartPolicy')
+
+    @property
+    def placement(self):
+        return self.get('Placement')
+
+
+class ContainerSpec(dict):
+    def __init__(self, image, command=None, args=None, env=None, workdir=None,
+                 user=None, labels=None, mounts=None, stop_grace_period=None):
+        from ..utils import split_command  # FIXME: circular import
+
+        self['Image'] = image
+
+        if isinstance(command, six.string_types):
+            command = split_command(command)
+        self['Command'] = command
+        self['Args'] = args
+
+        if env is not None:
+            self['Env'] = env
+        if workdir is not None:
+            self['Dir'] = workdir
+        if user is not None:
+            self['User'] = user
+        if labels is not None:
+            self['Labels'] = labels
+        if mounts is not None:
+            for mount in mounts:
+                if isinstance(mount, six.string_types):
+                    mounts.append(Mount.parse_mount_string(mount))
+                    mounts.remove(mount)
+            self['Mounts'] = mounts
+        if stop_grace_period is not None:
+            self['StopGracePeriod'] = stop_grace_period
+
+
+class Mount(dict):
+    def __init__(self, target, source, type='volume', read_only=False,
+                 propagation=None, no_copy=False, labels=None,
+                 driver_config=None):
+        self['Target'] = target
+        self['Source'] = source
+        if type not in ('bind', 'volume'):
+            raise errors.DockerError(
+                'Only acceptable mount types are `bind` and `volume`.'
+            )
+        self['Type'] = type
+
+        if type == 'bind':
+            if propagation is not None:
+                self['BindOptions'] = {
+                    'Propagation': propagation
+                }
+            if any(labels, driver_config, no_copy):
+                raise errors.DockerError(
+                    'Mount type is binding but volume options have been '
+                    'provided.'
+                )
+        else:
+            volume_opts = {}
+            if no_copy:
+                volume_opts['NoCopy'] = True
+            if labels:
+                volume_opts['Labels'] = labels
+            if driver_config:
+                volume_opts['driver_config'] = driver_config
+            if volume_opts:
+                self['VolumeOptions'] = volume_opts
+            if propagation:
+                raise errors.DockerError(
+                    'Mount type is volume but `propagation` argument has been '
+                    'provided.'
+                )
+
+    @classmethod
+    def parse_mount_string(cls, string):
+        parts = string.split(':')
```

Comment:
The CLI doesn't support this syntax for mounts, so I'm not sure it should be supported here